### PR TITLE
BaSP-TG-45: timeSheets

### PR DESCRIPTION
**Pull request by matescarabino**
- create list timeSheets with his update and delete buttons
- create a form to add new timeSheet and to update the previously listed ones
- add modals to show errors and double check on the delete function

### CODE REVIEW

**Review by Ansitox — COMMENTED** [review]
Just saw a wrong path on the layout, everything else looks fine

File: src/Components/Layout/index.js [hunk 1]
```diff
@@ -27,6 +28,9 @@ function Layout() {
     case '/time-sheets':
       currentScreen = <TimeSheets />;
       break;
+    case '/time-sheets/form':
+      currentScreen = <TimeSheetsFrom />;
```
[the remark this answers is not in the document]
Comment (ignacioaristo) [reply src/Components/Layout/index.js]: Done.

File: src/Components/TimeSheets/Form/index.js [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+        setTimeSheetInput({
+          description: json.data.description,
+          date: fixDate(json.data.date),
+          hours: json.data.hours,
+          task: json.data.task === null ? 'Not found in DB' : json.data.task['_id'],
+          employee: json.data.employee === null ? 'Not found in DB' : json.data.employee['_id'],
+          project: json.data.project === null ? 'Not found in DB' : json.data.project['_id']
+        });
+      } catch (error) {
+        console.error(error);
```
Comment (ignacioaristo) [inline src/Components/TimeSheets/Form/index.js:44]: Remove this console.log

File: src/Components/TimeSheets/index.js [hunk 3]
```diff
@@ -1,9 +1,112 @@
...
+      console.error(error);
+    }
+  }, []);
+
+  const closeModal = () => {
+    setShowModal(false);
+  };
+
+  const deleteTimeSheet = async (id) => {
+    saveList([...timeSheet.filter((timeSheet) => timeSheet._id !== id)]);
```
Comment (nicolasblasco) [inline src/Components/TimeSheets/index.js:25]: We are filtering the list by the deleted timesheet, what if the request fails? The timesheet won't be deleted from the database but the view will have a list without that timesheet.

File: src/Components/Layout/index.js [hunk 4]
```diff
@@ -27,6 +28,9 @@ function Layout() {
     case '/time-sheets':
       currentScreen = <TimeSheets />;
       break;
+    case '/time-sheets/form':
```
Comment (nicolasblasco) [inline src/Components/Layout/index.js:31]: Timesheet is written together. So the route should be named /timesheets/form

File: src/Components/Layout/index.js [hunk 5]
```diff
@@ -27,6 +28,9 @@ function Layout() {
     case '/time-sheets':
       currentScreen = <TimeSheets />;
```
Comment (nicolasblasco) [inline src/Components/Layout/index.js:29]: Timesheet is written together. So the route should be named /timesheets

**Review by nicolasblasco — DISMISSED** [review]

**Review by ignacioaristo — CHANGES_REQUESTED** [review]
Change all the components to arrow function.

File: src/Components/TimeSheets/Form/index.js (hunk 2 above)
Comment (matescarabino) [reply src/Components/TimeSheets/Form/index.js:44]: Done

File: src/Components/TimeSheets/index.js (hunk 3 above)
Comment (matescarabino) [reply src/Components/TimeSheets/index.js:25]: Done

File: src/Components/Layout/index.js (hunk 5 above)
Comment (matescarabino) [reply src/Components/Layout/index.js:29]: It was like that on the base file but it makes sense. Done

File: src/Components/TimeSheets/index.js [hunk 6]
```diff
@@ -1,9 +1,112 @@
...
+  const [showModal, setShowModal] = useState(false);
+  const [timeSheetId, setTimeSheetId] = useState(undefined);
+
+  useEffect(async () => {
+    try {
+      const response = await fetch(`${process.env.REACT_APP_API_URL}timesheets/`);
+      const json = await response.json();
+      saveList(json.data);
+    } catch (error) {
+      console.error(error);
```
[the remark this answers is not in the document]
Comment (matescarabino) [reply src/Components/TimeSheets/index.js:16]: Done

File: src/Components/TimeSheets/index.js [hunk 7]
```diff
@@ -1,9 +1,112 @@
+import { useEffect, useState } from 'react';
+import Modal from './Modal/index';
 import styles from './time-sheets.module.css';
 
 function TimeSheets() {
```
[the remark this answers is not in the document]
Comment (matescarabino) [reply src/Components/TimeSheets/index.js:5]: Done

File: src/Components/TimeSheets/index.js [hunk 8]
```diff
@@ -1,9 +1,112 @@
...
+  const [timeSheet, saveList] = useState([]);
```
[the remark this answers is not in the document]
Comment (matescarabino) [reply src/Components/TimeSheets/index.js:6]: Done

File: src/Components/TimeSheets/Form/Form Title/index.js [hunk 9]
```diff
@@ -0,0 +1,5 @@
+function FormText(props) {
```
Comment (ignacioaristo) [inline src/Components/TimeSheets/Form/Form Title/index.js:1]: Is this component necessary? I think we just could put the title in the index.js of the form.

**Review by ignacioaristo — CHANGES_REQUESTED** [review]
The name of the folder must be with cameCasel convention.

**Review by ignacioaristo — APPROVED** [review]

**Review by renzopimentel — APPROVED** [review]
Seems to be everything ok, approved 👌🏻

**Review by MaxiG-dev — APPROVED** [review]
Everything looks great!

**Review by julianpiaggio — COMMENTED** [review]
🕵️‍♂️ **QA FEEDBACK**

**TIME SHEETS**

First of all, we wanted to tell you that we consider that your pr complies
the requirements to be approved, and that you can merge to the master branch 👍

Now, if we consider leaving you some comments:

* in the create action, the description field allows only spaces, only numbers, and only characters.
 What we consider necessary would be to be able to add a validation that yes or yes contains letters and that
these may be accompanied by other characters.

* in the date field it allows to put numbers very far from the current one (eg: 01/01/0001). The advisable
it would be to put a minimum value so that it can be considered correct.

in the hours field, you can put 0 as a value, being inconsistent with the functionality of the field,
since it describes exactly how many hours are worked. It is recommended to put greater than 0

Below we leave details in images and a link that accesses a video to be able to reproduce and be more clear about the aforementioned details:

error timesheets

https://imgur.com/a/FBrzpAs

**Pull request merged**